src/test/java/org/jenkinsci/plugins/kubernetes/credentials/OpenShiftBearerTokenCredentialMockServer.java

```diff
@@ -0,0 +1,109 @@
+
+	private static void badLocationHandler(HttpExchange he) throws IOException {
+		String redirectURL = "bad";
+		he.getResponseHeaders().set("Location", redirectURL);
+		he.sendResponseHeaders(302, -1);
+	}
+
+	private static void missingLocationHandler(HttpExchange he) throws IOException {
+		he.sendResponseHeaders(302, -1); // No Location header
+	}
+
+	private static void badResponseHandler(HttpExchange he) throws IOException {
+		he.sendResponseHeaders(400, -1);
+	}
+
+	private static void validResponseHandler1(HttpExchange he) throws IOException {
+		String redirectURL = "http://my-service/#access_token=1234&expires_in=86400";
+		he.getResponseHeaders().set("Location", redirectURL);
+		he.sendResponseHeaders(302, -1);
+	}
+
+	private static void validResponseHandler2(HttpExchange he) throws IOException {
+		String redirectURL = "http://my-service/#access_token=1235&expires_in=86400";
+		he.getResponseHeaders().set("Location", redirectURL);
+		he.sendResponseHeaders(302, -1);
+	}
+
+	private static void defaultHandler(HttpExchange he) throws IOException {
+		String path = he.getRequestURI().getPath();
+		Pattern pattern = Pattern.compile("(.*)/.well-known/oauth-authorization-server");
+		Matcher matcher = pattern.matcher(path);
+
+		String scheme = "http";
+		if (he.getHttpContext().getServer() instanceof HttpsServer) {
+			scheme = "https";
+		}
+
+		String hostname = "localhost";
+		int port = he.getLocalAddress().getPort();
+
+		if (matcher.find()) {
+			String responseToClient = "{\n" + "  \"issuer\": \""
+			                          + scheme + "://" + hostname + ":" + port + "/\",\n" + "  \"authorization_endpoint\": \""
+			                          + scheme + "://" + hostname + ":" + port + matcher.group(1) + "/oauth/authorize\",\n"
+			                          + "  \"token_endpoint\": \""
+			                          + scheme + "://" + hostname + ":" + port + "/oauth/token\",\n" + "  \"scopes_supported\": [\n"
+			                          + "    \"user:check-access\",\n"
+			                          + "    \"user:full\",\n"
+			                          + "    \"user:info\",\n"
+			                          + "    \"user:list-projects\",\n"
+			                          + "    \"user:list-scoped-projects\"\n"
+			                          + "  ],\n"
+			                          + "  \"response_types_supported\": [\n"
+			                          + "    \"code\",\n"
+			                          + "    \"token\"\n"
+			                          + "  ],\n"
+			                          + "  \"grant_types_supported\": [\n"
+			                          + "    \"authorization_code\",\n"
+			                          + "    \"implicit\"\n"
+			                          + "  ],\n"
+			                          + "  \"code_challenge_methods_supported\": [\n"
+			                          + "    \"plain\",\n"
+			                          + "    \"S256\"\n"
+			                          + "  ]\n"
+			                          + "}";
```

Comment:
Could be great to update the plugin to Java 17 so you could use String Blocks here.
That would require to update `jenkins.version` and build configuration but might worth the change imo. What do you think @MarkEWaite ? 
Maybe for later ?

Reply:
When I tried to mix general purpose Java 17 upgrades with Spring Security 6.x Upgrade work it did not end well for me.    I tried to do that mixing with the git client plugin experiment with JGit 7.0.0 that requires Java 17.  I was stacking two patches (one for JGit 7.0.0 and requiring Jenkins 2.463 minimum version and the other requiring the Jelly 12 EE 9 pre-release from the tip of the jakarta branch) and the addition of the Java 17 changes made me discard both of those changes and start with fresh branches.

I think that in general we should avoid making general Java 17 changes so that we don't delay these pull requests that are time critical.

In this case, the plugin does not need to require Java 17, so the Java 17 changes would actually break the plugin.

Reply:
Totally fine by me. This can be a "cleanup" task later on when we know everything is ok with the plugin.